subcommands/targets/offline-update.go

```diff
@@ -163,6 +187,64 @@ func getTargetInfo(factory string, targetName string) (*ouTargetInfo, error) {
 	return &info, nil
 }
 
+func getWaveTargetMeta(factory string, targetName string, wave string) (*tuf.FileMeta, error) {
+	waveTargets, err := api.WaveTargetsList(factory, true, wave)
+	if err != nil {
+		if herr := client.AsHttpError(err); herr != nil && herr.Response.StatusCode == 404 {
+			return nil, fmt.Errorf("no active Wave with the specified name was found; wave: %s", wave)
+		}
+		return nil, fmt.Errorf("failed to get Wave Target metadata: %s", err.Error())
+	}
+	targets, ok := waveTargets[wave]
+	if !ok {
+		// it looks rather like panic since the call at #281 is supposed to return only the given wave targets
+		subcommands.DieNotNil(fmt.Errorf("no Targets are found for the given wave %s", wave))
+	}
```

Comment:
Just IMHO you're paying to much attention to these errors.
I'd just return `if err != nil { return nil, err }` at line 192 and be done.

At the least, lines 198-202 indeed try to handle an impossible use case... we're kind of not trusting our own API here.  Looks weird.

Reply:
>I'd just return if err != nil { return nil, err } at line 192 and be done.

I did so in the first place. The error message printed by `fioctl` is really confusing. I don't understand it. So. those `if err != nil { if herr := client.AsHttpError(err);` are intended for printing understandable error message.


>At the least, lines 198-202 indeed try to handle an impossible use case... we're kind of not trusting our own API here. Looks weird.

So, just panic?